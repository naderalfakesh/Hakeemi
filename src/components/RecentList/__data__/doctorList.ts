import { Specialty, Doctor } from '../../../services/api/types';
import firestore from '@react-native-firebase/firestore';

const list: Doctor[] = [
  {
    id: '1',
    avatar: 'https://images2.imgbox.com/f7/bb/7HJeSokt_o.png',
    title: Specialty.NEUROLOGY,
    name: '',
    specialty: [Specialty.NEUROLOGY],
    bio: '',
    email: '',
    address: '',
    phone: '',
    availability: [0, 1, 2],
    appointments: [
      {
        date: firestore.Timestamp.fromDate(new Date(2021, 10, 61)),
        userId: 'user id 1',
      },
    ],
  },
  {
    id: '2',
    avatar: 'https://images2.imgbox.com/5d/c9/sqb7cQMS_o.png',
    title: Specialty.DERMATOLOGY,
    name: '',
    specialty: [Specialty.NEUROLOGY],
    bio: '',
    email: '',
    address: '',
    phone: '',
    availability: [0, 1, 2],
    appointments: [],
  },
  {
    id: '3',
    avatar: 'https://images2.imgbox.com/19/80/QQZwXNbP_o.png',
    title: Specialty.PEDIATRICS,
    name: '',
    specialty: [Specialty.NEUROLOGY],
    bio: '',
    email: '',
    address: '',
    phone: '',
    availability: [3, 4, 5],
    appointments: [],
  },
  {
    id: '4',
    avatar: 'https://images2.imgbox.com/4b/fe/k8H0shP2_o.png',
    title: Specialty.SURGERY,
    name: '',
    specialty: [Specialty.NEUROLOGY],
    bio: '',
    email: '',
    address: '',
    phone: '',
    availability: [6],
    appointments: [],
  },
  {
    id: '5',
    avatar: 'https://imgbox.com/k8H0shP2',
    title: Specialty.PSYCHIATRY,
    name: '',
    specialty: [Specialty.NEUROLOGY],
    bio: '',
    email: '',
    address: '',
    phone: '',
    availability: [0, 2],
    appointments: [],
  },
  {
    id: '6',
    avatar: 'https://imgbox.com/lRBBYcMj',
    title: Specialty.UROLOGY,
    name: '',
    specialty: [Specialty.NEUROLOGY],
    bio: '',
    email: '',
    address: '',
    phone: '',
    availability: [0, 1, 2, 3, 4, 5, 6],
    appointments: [],
  },
  {
    id: '7',
    avatar: 'https://imgbox.com/h79FIvf2',
    title: Specialty.PEDIATRICS,
    name: '',
    specialty: [Specialty.NEUROLOGY],
    bio: '',
    email: '',
    address: '',
    phone: '',
    availability: [5],
    appointments: [],
  },
];

export default list;
