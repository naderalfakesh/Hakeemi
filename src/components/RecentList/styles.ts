import { StyleSheet } from 'react-native';
import spacing from '../../theme/spacing';

export default StyleSheet.create({
  base: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  list: {
    flexDirection: 'row',
  },
  avatar: {
    marginRight: spacing.small1,
  },
  extraButton: {
    paddingHorizontal: 21,
    paddingTop: 13,
  },
});
