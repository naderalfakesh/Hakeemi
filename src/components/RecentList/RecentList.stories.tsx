import React from 'react';
import { View, StyleSheet } from 'react-native';
import { storiesOf } from '@storybook/react-native';
import RecentList from './RecentList';
import doctorList from './__data__/doctorList';

const Container = (storyFn: any) => (
  <View style={styles.container}>{storyFn()}</View>
);

const RecentListDemo = () => (
  <RecentList
    doctors={doctorList}
    loading={false}
    onPress={id => console.log(id)}
    onExtraPress={() => console.log('extra pressed')}
  />
);

storiesOf('Recent list', module)
  .addDecorator(Container)
  .add('default', () => <RecentListDemo />);

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'rgba(251, 251, 251, 1)',
    flexGrow: 1,
    paddingHorizontal: 20,
    justifyContent: 'center',
  },
});
