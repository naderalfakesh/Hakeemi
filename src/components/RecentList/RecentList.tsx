import React, { FC, useState } from 'react';
import {
  View,
  ViewStyle,
  Image,
  StyleSheet,
  ActivityIndicator,
} from 'react-native';
import styles from './styles';
import Avatar from '../Avatar';
import Button from '../Button';
import { Doctor } from '../../services/api/types';
import colors from '../../theme/colors';

const AVATAR_SIZE = 50;
const ITEM_COUNT_TO_SHOW = 4;

interface Props {
  style?: ViewStyle;
  doctors: Array<Doctor>;
  onPress?: (id: string) => void;
  onExtraPress?: () => void;
  loading: boolean;
}

const RecentList: FC<Props> = ({
  style,
  doctors,
  onPress,
  onExtraPress,
  loading,
}) => {
  const [imageLoading, setImageLoading] = useState(true);

  return (
    <View testID="list" style={[styles.base, style]}>
      <View style={styles.list}>
        {doctors && doctors.length
          ? doctors.slice(0, ITEM_COUNT_TO_SHOW).map(doctor => (
              <Avatar
                key={doctor.id}
                theme="secondary"
                style={styles.avatar}
                onPress={() => onPress && onPress(doctor.id)}
                size={AVATAR_SIZE}
                shape="rounded"
                roundness={0.4}>
                <Image
                  resizeMode="contain"
                  width={AVATAR_SIZE}
                  height={AVATAR_SIZE}
                  fadeDuration={0}
                  source={{
                    uri: doctor.avatar,
                    width: AVATAR_SIZE,
                    height: AVATAR_SIZE,
                  }}
                  onLoad={() => setImageLoading(false)}
                />
                {imageLoading && (
                  <ActivityIndicator
                    style={StyleSheet.absoluteFill}
                    color={colors.primary}
                  />
                )}
              </Avatar>
            ))
          : undefined}
      </View>
      {doctors.length > ITEM_COUNT_TO_SHOW && (
        <Button
          loading={loading}
          style={styles.extraButton}
          onPress={() => onExtraPress && onExtraPress()}>
          {`+ ${doctors.length - ITEM_COUNT_TO_SHOW}`}
        </Button>
      )}
    </View>
  );
};

export default RecentList;
