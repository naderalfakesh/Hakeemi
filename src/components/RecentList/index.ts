import RecentList from './RecentList';

export default RecentList;
