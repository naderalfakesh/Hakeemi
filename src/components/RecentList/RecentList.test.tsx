import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import RecentList from './RecentList';
import doctorList from './__data__/doctorList';
jest.mock('@react-native-firebase/firestore', () => ({
  Timestamp: { fromDate: jest.fn },
}));

describe('Button component test', () => {
  it('renders correctly with image list', () => {
    const { toJSON } = render(
      <RecentList doctors={doctorList} loading={false} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with empty list', () => {
    const { toJSON } = render(<RecentList doctors={[]} loading={false} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('invokes a function when extra button is pressed', () => {
    const mockFn = jest.fn();
    const { getByTestId } = render(
      <RecentList doctors={doctorList} onExtraPress={mockFn} loading={false} />,
    );
    const btn = getByTestId('Button');
    fireEvent.press(btn);
    expect(mockFn).toBeCalled();
  });

  it('invokes a function when a doctor pressed', () => {
    const mockFn = jest.fn();
    const { getAllByTestId } = render(
      <RecentList doctors={doctorList} onPress={mockFn} loading={false} />,
    );
    const avatar = getAllByTestId('avatar');
    fireEvent.press(avatar[0]);
    expect(mockFn).toBeCalledWith('1');
  });
});
