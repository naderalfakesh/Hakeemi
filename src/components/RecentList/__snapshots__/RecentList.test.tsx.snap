// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test renders correctly with empty list 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
  testID="list"
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  />
</View>
`;

exports[`Button component test renders correctly with image list 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
  testID="list"
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#edf3f7",
            "borderRadius": 20,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          },
          undefined,
          Object {
            "marginRight": 10,
          },
        ]
      }
      testID="avatar"
    >
      <Image
        fadeDuration={0}
        height={50}
        onLoad={[Function]}
        resizeMode="contain"
        source={
          Object {
            "height": 50,
            "uri": "https://images2.imgbox.com/f7/bb/7HJeSokt_o.png",
            "width": 50,
          }
        }
        width={50}
      />
      <ActivityIndicator
        animating={true}
        color="#00bbd4"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#edf3f7",
            "borderRadius": 20,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          },
          undefined,
          Object {
            "marginRight": 10,
          },
        ]
      }
      testID="avatar"
    >
      <Image
        fadeDuration={0}
        height={50}
        onLoad={[Function]}
        resizeMode="contain"
        source={
          Object {
            "height": 50,
            "uri": "https://images2.imgbox.com/5d/c9/sqb7cQMS_o.png",
            "width": 50,
          }
        }
        width={50}
      />
      <ActivityIndicator
        animating={true}
        color="#00bbd4"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#edf3f7",
            "borderRadius": 20,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          },
          undefined,
          Object {
            "marginRight": 10,
          },
        ]
      }
      testID="avatar"
    >
      <Image
        fadeDuration={0}
        height={50}
        onLoad={[Function]}
        resizeMode="contain"
        source={
          Object {
            "height": 50,
            "uri": "https://images2.imgbox.com/19/80/QQZwXNbP_o.png",
            "width": 50,
          }
        }
        width={50}
      />
      <ActivityIndicator
        animating={true}
        color="#00bbd4"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#edf3f7",
            "borderRadius": 20,
            "height": 50,
            "overflow": "hidden",
            "width": 50,
          },
          undefined,
          Object {
            "marginRight": 10,
          },
        ]
      }
      testID="avatar"
    >
      <Image
        fadeDuration={0}
        height={50}
        onLoad={[Function]}
        resizeMode="contain"
        source={
          Object {
            "height": 50,
            "uri": "https://images2.imgbox.com/4b/fe/k8H0shP2_o.png",
            "width": 50,
          }
        }
        width={50}
      />
      <ActivityIndicator
        animating={true}
        color="#00bbd4"
        hidesWhenStopped={true}
        size="small"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 16,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#00bbd4",
        },
        Object {
          "padding": 16,
        },
        false,
        false,
        Object {
          "paddingHorizontal": 21,
          "paddingTop": 13,
        },
      ]
    }
    testID="Button"
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
          "includeFontPadding": false,
          "lineHeight": 20,
          "textAlign": undefined,
        }
      }
      testID="Text"
    >
      + 3
    </Text>
  </View>
</View>
`;
